keeper/src/main/scala/zio/keeper/hyparview/Message.scala

```diff
@@ -78,114 +100,85 @@ object Message {
 
   }
 
-  final case class Join(
-    sender: NodeAddress
-  ) extends Message
-
-  object Join {
-
-    implicit val codec: ByteCodec[Join] =
-      ByteCodec.fromReadWriter(macroRW[Join])
-  }
-
-  final case class ForwardJoinReply(
-    sender: NodeAddress
-  ) extends Message
+  final case class Shuffle(
+    sender: NodeAddress,
+    originalSender: NodeAddress,
+    activeNodes: List[NodeAddress],
+    passiveNodes: List[NodeAddress],
+    ttl: TimeToLive
+  ) extends ViewMessage
 
-  object ForwardJoinReply {
+  object Shuffle {
 
-    implicit val codec: ByteCodec[ForwardJoinReply] =
-      ByteCodec.fromReadWriter(macroRW[ForwardJoinReply])
+    implicit val codec: ByteCodec[Shuffle] =
+      ByteCodec.fromReadWriter(macroRW[Shuffle])
   }
 
   final case class ShuffleReply(
     passiveNodes: List[NodeAddress],
     sentOriginally: List[NodeAddress]
-  ) extends Message
+  ) extends ViewMessage
 
   object ShuffleReply {
 
     implicit val codec: ByteCodec[ShuffleReply] =
       ByteCodec.fromReadWriter(macroRW[ShuffleReply])
   }
 
-  final case class JoinReply(
-    remote: NodeAddress
-  ) extends Message
-
-  object JoinReply {
+  /**
+   * Messages used by plumtree layer.
+   */
+  sealed abstract class PeerMessage extends ViewMessage
 
-    implicit val codec: ByteCodec[JoinReply] =
-      ByteCodec.fromReadWriter(macroRW[JoinReply])
+  case object Prune extends PeerMessage {
```

Comment:
Let's put terms in companion.

Reply:
There is trait hierarchy here that encodes what can be done with different subsets of messages. They already live in the Message companion object, but nesting them further makes usability really bad.

i.e. `Message.PeerMessage.Prune`